File: month02_llm_engineering/week05_llm_fundamentals/day34_embeddings/node/embedding.js
import { HfInference } from "@huggingface/inference";
import dotenv from "dotenv";

dotenv.config({
  path: "../.env",
});

const hf = new HfInference(process.env.HF_TOKEN);

async function generateEmbedding(text) {
  try {
    const result = await hf.featureExtraction({
      model: "sentence-transformers/all-MiniLM-L6-v2",
      inputs: text,
    });

    console.log("Dimensions:", result.length);
    console.log("First 10 values:");
    console.log(result.slice(0, 10));

    return result;

  } catch (error) {
    console.error(error.message);
  }
}

generateEmbedding(
  "Node.js is a JavaScript runtime."
);

File: month02_llm_engineering/week05_llm_fundamentals/day34_embeddings/python/embedding.py
import os
from dotenv import load_dotenv
from huggingface_hub import InferenceClient


load_dotenv("../.env")


HF_TOKEN = os.getenv("HF_TOKEN")


client = InferenceClient(
    token=HF_TOKEN
)


def generate_embedding(text):

    result = client.feature_extraction(
        text,
        model="sentence-transformers/all-MiniLM-L6-v2"
    )


    embedding = result.tolist()


    print("Dimensions:", len(embedding))

    print("First 10 values:")
    print(embedding[:10])


    return embedding



generate_embedding(
    "Node.js is a JavaScript runtime."
)